nixd/lib/Controller/Hover.cpp

```diff
@@ -22,6 +22,31 @@ using namespace lspserver;
 
 namespace {
 
+class OptionsHoverProvider {
+  AttrSetClient &Client;
+
+public:
+  OptionsHoverProvider(AttrSetClient &Client) : Client(Client) {}
+  std::optional<OptionDescription> resolveHover(std::vector<std::string> Scope,
+                                                std::string Name) {
```

Comment:
you will not need this `Name` parameter. The `Scope` itself will be necessary for option information.